data/src/main/java/me/androidbox/data/remote/interceptor/TokenInterceptor.kt

```diff
@@ -0,0 +1,28 @@
+package me.androidbox.data.remote.interceptor
+
+import me.androidbox.domain.authentication.preference.PreferenceRepository
+import okhttp3.Interceptor
+import okhttp3.Response
+
+class TokenInterceptor(private val preferenceRepository: PreferenceRepository) : Interceptor {
+
+    private companion object {
+        const val AUTHORIZATION_HEADER = "Authorization"
+    }
+
+    override fun intercept(chain: Interceptor.Chain): Response {
+        val loginUser = preferenceRepository.retrieveCurrentUserOrNull()
+
+        return if(loginUser != null) {
+            val request = chain.request()
+                .newBuilder()
+                .addHeader(AUTHORIZATION_HEADER, "Bearer ${loginUser.token}")
+                .build()
+
+            chain.proceed(request)
+        }
+        else {
+            chain.proceed(chain.request())
+        }
+    }
```

Comment:
This is my personal opinion:

I usually prefer to not have != if later I'm going to add an else

So right now:

if (loginUser!= null) {} else {}

I prefer having a == null and then an else. 

So I would do:

```

return if (loginUser == null) {
    chain.proceed(chain.request())
} else {
    val request = chain.request()
                .newBuilder()
                .addHeader(AUTHORIZATION_HEADER, "Bearer ${loginUser.token}")
                .build()
    chain.proceed(request)
}
```

Reply:
Updated the to check for == null in the if statement